morf-core/src/main/java/org/alfasoftware/morf/upgrade/Upgrade.java

```diff
@@ -327,4 +326,41 @@ private Schema copySourceSchema(ConnectionResources database, DataSource dataSou
       databaseSchemaResource.close();
     }
   }
+
+
+  /**
+   * Factory that can be used to create {@link Upgrade}s.
+   *
+   * @author Copyright (c) Alfa Financial Software 2022
+   */
+  public static class Factory  {
+    private final UpgradePathFactory upgradePathFactory;
+    private final GraphBasedUpgradeBuilderFactory graphBasedUpgradeBuilderFactory;
+    private final UpgradeStatusTableService.Factory upgradeStatusTableServiceFactory;
+    private final ViewChangesDeploymentHelper.Factory viewChangesDeploymentHelperFactory;
+    private final ViewDeploymentValidator.Factory viewDeploymentValidatorFactory;
+
+
+    @Inject
+    public Factory(UpgradePathFactory upgradePathFactory,
+                   UpgradeStatusTableService.Factory upgradeStatusTableServiceFactory,
+                   GraphBasedUpgradeBuilderFactory graphBasedUpgradeBuilderFactory,
+                   ViewChangesDeploymentHelper.Factory viewChangesDeploymentHelperFactory,
+                   ViewDeploymentValidator.Factory viewDeploymentValidatorFactory) {
+      this.upgradePathFactory = upgradePathFactory;
+      this.graphBasedUpgradeBuilderFactory = graphBasedUpgradeBuilderFactory;
+      this.upgradeStatusTableServiceFactory =  upgradeStatusTableServiceFactory;
+      this.viewChangesDeploymentHelperFactory = viewChangesDeploymentHelperFactory;
+      this.viewDeploymentValidatorFactory = viewDeploymentValidatorFactory;
+    }
+
+    public Upgrade create(ConnectionResources connectionResources) {
+      return new Upgrade(connectionResources,
+                         upgradePathFactory,
+                         upgradeStatusTableServiceFactory.create(connectionResources),
+                         viewChangesDeploymentHelperFactory.create(connectionResources),
+                         viewDeploymentValidatorFactory.createViewDeploymentValidator(connectionResources),
+                         graphBasedUpgradeBuilderFactory);
+    }
+  }

```

Comment:
Injection noticeable change:
Switch to this Upgrade factory requires switch to binding new factories.